### PR TITLE
add msvc mmap

### PR DESCRIPTION
**Pull request by yajiedesign**
## Description ##
add msvc mmap.wait unittest.

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] For user-facing API changes, API doc string has been updated.
- [ ] To my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Intersting edge cases to note here

### CODE REVIEW

File: include/mxnet/storage.h [hunk 1]
```diff
@@ -55,6 +55,9 @@ class Storage {
      */
     int shared_pid{-1};
     int shared_id{-1};
+#ifdef _WIN32
+    void* map_handle{nullptr};
```
Comment (piiswrong) [inline include/mxnet/storage.h]: keep this inside manager in a map?

File: src/storage/cpu_shared_storage_manager.h [hunk 2]
```diff
@@ -94,9 +100,17 @@ class CPUSharedStorageManager final : public StorageManager {
   std::mutex mutex_;
   std::mt19937 rand_gen_;
   std::unordered_map<void*, Storage::Handle> pool_;
+#ifdef _WIN32
+  std::list<Storage::Handle> wait_real_free_;
```
Comment (piiswrong) [inline src/storage/cpu_shared_storage_manager.h]: why not use pool?
